src/main/java/com/avanza/gs/test/junit5/ResourceExtension.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017 Avanza Bank AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.avanza.gs.test.junit5;
+
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.ExtensionContext.Store;
+
+interface ResourceExtension extends BeforeAllCallback, BeforeEachCallback {
```

Comment:
Here is a suggestion for javadoc for this class:
```java
/**
 * This interface makes sure that {@link #before} and {@link #after} are called
 * at most once, regardless of whether the implementing class is used in tests
 * as a {@code static} instance (which would normally use the callbacks from
 * {@link BeforeAllCallback}), or as a non-{@code static} instance (which would
 * normally use {@link BeforeEachCallback}).
 */
```

Reply:
Awesome, I've added it! 👍